network/p2p/scoring/registry_test.go

```diff
@@ -422,6 +423,63 @@ func TestPersistingInvalidSubscriptionPenalty(t *testing.T) {
 	assert.Equal(t, 0.0, record.Penalty) // penalty should be zero.
 }
 
+// TestPeerSpamPenaltyClusterPrefixed evaluates the application-specific penalty calculation for a node when a spam record is present
+// for cluster-prefixed topics. In the case of an invalid control message notification marked as cluster-prefixed,
+// the application-specific penalty should be reduced by the default reduction factor. This test verifies the accurate computation
+// of the application-specific score under these conditions.
+func TestPeerSpamPenaltyClusterPrefixed(t *testing.T) {
```

Comment:
This test only concerns `GRFAT` control message types; please extend it to support other message types as well in the same scenario.

Reply:
https://github.com/onflow/flow-go/pull/4857/commits/1749d789e61dff2384e7ad8ef332a9bda642bd01